src/odemis/acq/stream/_sync.py

```diff
@@ -1787,7 +1790,220 @@ def _assembleSpecData(self, data_list, repetition):
+                self._anchor_raw.append(self._assembleAnchorData(drift_est.raw))
+            if self._acq_state == CANCELLED:
+                raise CancelledError()
+        except CancelledError:
+            logging.info("Time correlator stream cancelled")
+            with self._acq_lock:
+                self._acq_state = FINISHED
+            raise  # Just don't log the exception
+        except Exception:
+            logging.exception("Failure during Correlator acquisition")
+            raise
+        else:
+            return self.raw
+        finally:
+            logging.debug("TC acquisition finished")
+            self._acq_done.set()
+            # Reset hardware settings (dwell times might have been reduced due to subpixel drift
+            # correction
+            dwell_time = self._tc_stream.dwellTime.value * max(1, self._tc_stream.nDC.value)
+            self._tc_stream._detector.dwellTime.value = dwell_time
+            self._emitter.dwellTime.value = dwell_time
+    
+    def _acquireImage(self, x, y):
+        try: 
+            for ce in self._acq_complete:
+                ce.clear()
+            
+            self._emitter.translation.value = (x, y)
+            # checks the hardware has accepted it
+            trans = self._emitter.translation.value
+            if math.hypot(x - trans[0], y - trans[1]) > 1e-3:
+                logging.warning("Ebeam translation is %s instead of requested %s.", trans, (x, y))
+            
+            # Get data
+            for s, sub in zip(self._streams, self._subscribers):
+                s._dataflow.subscribe(sub)
+            
+            # Wait for detector to acquire image
+            dt = self._tc_stream.dwellTime.value * 1.5 + 1
+            for i, s in enumerate(self._streams):
+                if not self._acq_complete[i].wait(dt):
+                    raise TimeoutError()
+            if self._acq_state == CANCELLED:
+                raise CancelledError()
+            tc_data, se_data = self._acq_data[-1][-1], self._acq_data[0][-1]
+            return tc_data, se_data
+        finally:
+            for s, sub in zip(self._streams, self._subscribers):
+                s._dataflow.unsubscribe(sub)   
+    
+    def _onCompletedData(self, n, raw_das):        
+        md = raw_das[0].metadata.copy()
+        md[MD_DESCRIPTION] = self._streams[n].name.value
+        md[model.MD_PIXEL_SIZE] = (self._tc_stream.pixelSize.value, self._tc_stream.pixelSize.value)
+        md[model.MD_DWELL_TIME] = self._tc_stream.dwellTime.value
+        md[model.MD_DIMS] = "CTZYX"
+        # For now, the viewport cannot display large datasets, so we have to crop the temporal data
+        if n == 1:
+            for i, da in enumerate(raw_das):
+                raw_das[i] = da[:, :self._tc_stream.cropvalue.value]
+
+        shape = numpy.array(raw_das).shape
+        rep = self._tc_stream.repetition.value
+        das = numpy.reshape(raw_das, [shape[2], 1, 1, rep[1], rep[0]])
```

Comment:
Please indicate why this works, both for SEM and time-correlator: the SEM data is of shape 1,1 (YX), while the time-correlator data is of shape 1, 65535 (XT). So the first dimension can always be discarded and the second dimension can be considered T.

Note that if we were supporting fuzzing, this wouldn't work anymore, and we'd need to do the same as the rest of the code, which has a "standard" function for SEM, and specialised version for the CCD.